lib/rubocop/cop/rails/env.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks typo of environment name.
+      #
+      # @example
+      #   # bad
+      #   Rails.env.proudction?
+      #
+      #   # good
+      #   Rails.env.production?
+      class Env < Cop
+        include NameSimilarity
+
+        MSG = '`%<name>s` is incorrect environment name.'.freeze
+        MSG_SIMILAR = '`%<name>s` is incorrect environment name. ' \
+                      'Did you mean `%<similar>s`?'.freeze
+
+        def_node_matcher :typo_environment?, <<-PATTERN
+          (send
+            (send
+              {(const nil :Rails) (const (cbase) :Rails)}
+              :env)
+            $#incorrect_env_name?)
+        PATTERN
+
+        def on_send(node)
+          typo_environment?(node) do |name|
+            add_offense(node, :selector, message(name))
+          end
+        end
+
+        private
+
+        def collect_variable_like_names(_scope)
+          environments
+        end
+
+        def message(name)
+          similar = find_similar_name(name.to_s, [])
+          if similar
+            format(MSG_SIMILAR, name: name, similar: similar)
+          else
+            format(MSG, name: name)
+          end
+        end
+
+        def incorrect_env_name?(name)
+          name = name.to_s
+          name.end_with?('?') &&
+            !environments.include?(name)
+        end
+
+        def environments
+          cop_config['Environments'].map { |env| env + '?' }
```

Comment:
Technically the `?` isn't part of the environment name. Might want to strip it when before checking `include?` instead.